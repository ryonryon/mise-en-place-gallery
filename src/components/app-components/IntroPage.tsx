import styled from "@emotion/styled";
import * as React from "react";

interface Props {
  title: string;
  description: string;
  children: React.ReactNode;
  imgSrcs: [string, string][] | null;
}

const StyledDiv = styled.div`
  width: 100%;
  text-align: center;
  min-height: 350px;
  position: relative;

  .pane {
    display: flex;
    flex-flow: row nowrap;

    .left-pane,
    .right-pane {
      display: inline-block;
      width: 50%;
      margin: 0 auto;
    }

    .left-pane {
      p {
        line-height: 150%;
      }
    }

    .right-pane {
      padding-top: 50px;
    }
  }

  & > hr {
    position: relative;
    bottom: 0;
  }
`;

const IntroPage = ({ title, description, children, imgSrcs }: Props) => (
  <StyledDiv>
    <div className="pane" id={title.toLowerCase()}>
      <div className="left-pane">
        <h2>{title}</h2>
        <p>{description}</p>
      </div>
      <div className="right-pane">{children}</div>
    </div>
    <>
      {imgSrcs.map(([imgSrc, imgTag], index) => (
        <div key={index}>
          <img src={imgSrc} alt={imgTag} />
          <p>{imgTag}</p>
        </div>
      ))}
    </>
    <hr />
  </StyledDiv>
);

IntroPage.defaultProps = {
  imgSrcs: []
};

export default IntroPage;
