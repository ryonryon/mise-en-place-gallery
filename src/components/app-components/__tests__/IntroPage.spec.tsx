import * as React from "react";
import * as renderer from "react-test-renderer";

import IntroPage from "../IntroPage";

describe("IntroPage test", () => {
  it("tests IntroPage renders correctly ", () => {
    const introPage = renderer
      .create(
        <IntroPage title="test" description="test">
          test
        </IntroPage>
      )
      .toJSON();
    expect(introPage).toMatchSnapshot();
  });
});
