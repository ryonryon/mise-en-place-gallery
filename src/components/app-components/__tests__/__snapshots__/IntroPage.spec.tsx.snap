// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroPage test tests IntroPage renders correctly  1`] = `
<div
  className="css-svh4v8"
>
  <div
    className="pane"
    id="test"
  >
    <div
      className="left-pane"
    >
      <h2>
        test
      </h2>
      <p>
        test
      </p>
    </div>
    <div
      className="right-pane"
    >
      test
    </div>
  </div>
  <hr />
</div>
`;
